### PR TITLE
Selector Signal Interoperability.

### PR DESCRIPTION
**Pull request by andersio**
Included test cases from #27.

#### Terminology
Stated class: The class the instance meant to be at compile time.
Runtime subclass: The subclass of the perceived class created at runtime.

#### Summary
RAC now respects any further swizzling of `forwardInvocation:`, `responseToSelector:` and method implementations **made to the stated class**.

The PR also fixed a bug that caused an unrecognised selector exception when calling a setter that has been swizzled by KVO and subsequently by RAC.

#### Implementation Notes
Before the PR, RAC grabs the IMP of any selector by _searching through the inheritance hierarchy_, and aliases it or preserves it before replacing the IMP with `_objc_msgForward` or its custom implementation.

After the PR, RAC _generally_ does not alias any IMP, and relies on the ObjC runtime at the call time to resolve & call the method with an IMP swapping trick.

#### What's not solved
Interoperability with frameworks (except Cocoa KVO) that isa-swizzle an instance and provide custom implementations for the following selectors **_in the runtime subclass_**: `forwardInvocation:`, `methodSignatureForSelector:`, and `respondsToSelector:`.

Related: https://github.com/ReactiveCocoa/ReactiveCocoa/issues/1561

### CODE REVIEW

File: ReactiveObjC/NSObject+RACSelectorSignal.m [hunk 1]
```diff
@@ -75,26 +52,47 @@ static void RACSwizzleForwardInvocation(Class class) {
...
+		RACSubject* subject = objc_getAssociatedObject(self, aliasSelector);
+
+		BOOL responseToSelector = [superclass instancesRespondToSelector:originalSelector];
+		BOOL forward = NO;
+		if (responseToSelector) {
+			Method method = class_getInstanceMethod(superclass, invocation.selector);
+			IMP impl = method_getImplementation(method);
+
+			if (impl != _objc_msgForward) {
+				class_replaceMethod(object_getClass(self), aliasSelector, impl, method_getTypeEncoding(method));
```
Comment (andersio) [inline ReactiveObjC/NSObject+RACSelectorSignal.m]: This looks nasty, but there is no reliable way to invoke a specific IMP with `NSInvocation` in the standard toolbox.

**Comment by andersio** [comment]
Hmm, not ready yet. Had a problem with KVO.

**Comment by andersio** [comment]
It seems there is just a slight performance regression.

[The test](https://gist.github.com/andersio/530cfa5a314d2b13b28ec935fc352fac)

100000 iterations
`selector-signal-interop`: 15.959s (43%)
`master`: 15.710s (49%)

**Comment by sprynmr** [comment]
What's the state of this PR?

**Comment by doggy** [comment]
@sprynmr Still working in testing progress

**Comment by andersio** [comment]
Gonna port back a few changes from the Swift version.

**Pull request closed without merging**